generators/swift/codegen/src/ast/Type.ts

```diff
@@ -0,0 +1,127 @@
+import { AstNode, Writer } from "@fern-api/generator-commons";
+import Swift, { AccessLevel, ClassLevel, Enum, Func } from "..";
+import { Field } from "./Field";
+
+/*
+
+Builds Swift Types (Classes, Structs, etc)
+==========================================
+
+Example:
+
+private class Mike: Person {
+
+    class Snowboard {
+        ...
+    }
+    
+    func wave() {
+        print("👋")
+    }
+
+}
+
+Breakdown:
+
+{accessLevel} {classLevel} {name}: {inheritance} {
+
+    {subclasses}
+    
+    {functions}
+
+}
+
+*/
+
+export declare namespace Type {
+    interface Args {
+        /* The access level of the type */
+        accessLevel?: AccessLevel;
+        /* The class level of the type, such as class or struct */
+        classLevel?: ClassLevel;
+        /* The name of the type */
+        name: string;
+        /* The subclasses of this type, which can be other types or enums */
+        subclasses?: (Type | Enum)[];
+        /* The field variables in the class */
+        fields?: Field[];
+        /* The functions associated with this type */
+        functions?: Func[];
+        /* The inheritance hierarchy of this type */
+        inheritance?: Type[];
+    }
+}
+
+export class Type extends AstNode {
```

Comment:
nit: as a follow up i think its worth calling this `Class` for clarity (already talked about)